### PR TITLE
Spalloc server jdbc

### PR DESCRIPTION
**Pull request by rowleya**
Fixes to the spalloc client to make it stop correctly.

Part of SpiNNakerManchester/JavaSpiNNaker#737 (but can be merged without that as only useful when using Java spalloc)

### CODE REVIEW

File: spinnman/spalloc/spalloc_client.py [hunk 1]
```diff
@@ -637,7 +715,7 @@ def _receive(self, timeout=None) -> bytes:
 
     def _is_ready_to_receive(self, timeout=0) -> bool:
         # If we already have a message or the queue peek succeeds, return now
-        if self.__current_msg is not None or self.__msgs.not_empty:
+        if self.__current_msg is not None or self.__msgs.qsize():
```
Comment (dkfellows) [inline spinnman/spalloc/spalloc_client.py:718]: Better as `not self.__msgs.empty()`, if my reading of the docs is right. We don't care how many items are in the queue, just whether there is anything.

**Review by dkfellows — APPROVED** [review]
Minor quibble (and what you've got is definitely better than what was there).

Thinking a bit more (and having actually read the [queue library docs](https://docs.python.org/3/library/queue.html) properly): perhaps the `__msgs` ought to be a `queue.SimpleQueue` instead? That's where the extra performance would be welcome; it's much less critical in the synchronous call handling.

**Comment by dkfellows** [comment]
The minimal features of the Java code to support this are in `master` over there and have been for ages. This can be handled on its own terms independent of SpiNNakerManchester/JavaSpiNNaker#737

File: spinnman/spalloc/spalloc_client.py (hunk 1 above)
Comment (rowleya) [reply spinnman/spalloc/spalloc_client.py]: Fixed

**Comment by rowleya** [comment]
Updates done.  This can indeed be merged independently of the Java branch.

**Pull request merged**